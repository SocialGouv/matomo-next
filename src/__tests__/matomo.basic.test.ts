import "./test-setup";
import { initPagesRouter, push } from "..";

describe("Basic Matomo functionality", () => {
  beforeEach(() => {
    global._paq = [];
  });

  describe("initPagesRouter", () => {
    it("should create a js tag and initialize", () => {
      document.head.appendChild(document.createElement("script"));
      initPagesRouter({ siteId: "42", url: "https://YO" });
      expect(global._paq).toMatchSnapshot();
    });

    it("should NOT create events when url is not provided", () => {
      document.head.appendChild(document.createElement("script"));
      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
      // @ts-expect-error
      initPagesRouter({ siteId: "42" });
      expect(global._paq).toMatchSnapshot();
    });
  });

  describe("push", () => {
    test("should append data to window._paq", () => {
      initPagesRouter({ siteId: "42", url: "YO" });
      window._paq = [];
      push(["trackEvent", "kikoo", "lol"]);
      expect(window._paq).toMatchSnapshot();
    });

    test("should append dimensions data to window._paq", () => {
      initPagesRouter({ siteId: "42", url: "YO" });
      window._paq = [];
      push([
        "trackEvent",
        "kikoo",
        "lol",
        null,
        null,
        { dimension1: "ok", dimension2: "foobar" },
      ]);
      expect(window._paq).toMatchSnapshot();
    });
  });

  describe("onInitialization", () => {
    test("should work if the surcharge of the operator", () => {
      initPagesRouter({
        onInitialization: () => {
          push(["during_initialization", "hello"]);
        },
        siteId: "42",
        url: "YO",
      });
      expect(window._paq).toEqual(
        expect.arrayContaining([["during_initialization", "hello"]]),
      );
    });
  });
});
