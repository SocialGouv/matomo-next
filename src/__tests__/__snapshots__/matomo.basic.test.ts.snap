// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic Matomo functionality initPagesRouter should NOT create events when url is not provided 1`] = `Array []`;

exports[`Basic Matomo functionality initPagesRouter should create a js tag and initialize 1`] = `
Array [
  Array [
    "trackPageView",
  ],
  Array [
    "enableLinkTracking",
  ],
  Array [
    "setTrackerUrl",
    "https://YO/matomo.php",
  ],
  Array [
    "setSiteId",
    "42",
  ],
]
`;

exports[`Basic Matomo functionality push should append data to window._paq 1`] = `
Array [
  Array [
    "trackEvent",
    "kikoo",
    "lol",
  ],
]
`;

exports[`Basic Matomo functionality push should append dimensions data to window._paq 1`] = `
Array [
  Array [
    "trackEvent",
    "kikoo",
    "lol",
    null,
    null,
    Object {
      "dimension1": "ok",
      "dimension2": "foobar",
    },
  ],
]
`;
